### PR TITLE
check_rdns: update cf_at_root logic

### PR DESCRIPTION
**Pull request by RhinosF1**
use a cleaner list and replace some Cloudflare specific naming with generic stuff.

### CODE REVIEW

**Comment by RhinosF1** [comment]
@paladox: as requested.

File: modules/monitoring/files/check_reverse_dns.py [hunk 1]
```diff
@@ -78,8 +78,14 @@ def check_records(hostname):
         for nameserver in nameserversans:
             nameserver = str(nameserver)
             nameservers.append(nameserver)
-            if nameserver.endswith('.ns.cloudflare.com.') or nameserver.endswith('.dreamhost.com.') or nameserver.endswith('.ns.porkbun.com.'):
-                uses_cf_at_root = True
+            flatten_manadatory_providers = [
+                'ns.cloudlfare.com',
+                'dreamhost.com',
+                'porkbun.com',
```
Comment (Universal-Omega) [inline modules/monitoring/files/check_reverse_dns.py:84]:
```suggestion
                'ns.porkbun.com',
```
For what it was before anyway.

File: modules/monitoring/files/check_reverse_dns.py [hunk 2]
```diff
@@ -78,8 +78,14 @@ def check_records(hostname):
...
+            ]
+            for provider in flatten_manadatory_providers:
+                if nameserver.endswith(f'.{provider}'):
```
Comment (Universal-Omega) [inline modules/monitoring/files/check_reverse_dns.py:87]:
```suggestion
                if nameserver.endswith(f'.{provider}.'):
```

Comment (Universal-Omega) [reply modules/monitoring/files/check_reverse_dns.py:87]: Also, endswith supports a tuple I think, so if list above is converted to a tuple, we do not need the loop I think?

https://docs.python.org/3/library/stdtypes.html#str.endswith

> suffix can also be a tuple of suffixes to look for.

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

File: modules/monitoring/files/check_reverse_dns.py (hunk 2 above)
Comment (RhinosF1) [reply modules/monitoring/files/check_reverse_dns.py:87]: No reason why I can't make it a tuple

Comment (Universal-Omega) [reply modules/monitoring/files/check_reverse_dns.py:87]: This should be enough here also, to just add minor simplification I think, but that is up to you.
```lang=py
if nameserver.endswith(('.ns.cloudlfare.com.', '.dreamhost.com.', '.ns.porkbun.com.')):
    cname_check_impossible = True
```

Comment (RhinosF1) [reply modules/monitoring/files/check_reverse_dns.py]: It's not simpler imho. Moving it to a tuple outside with each entry on a new line looks cleaner.

Comment (Universal-Omega) [reply modules/monitoring/files/check_reverse_dns.py]: Feel free to do however you feel best then.

**Review by Universal-Omega — APPROVED** [review]

**Pull request merged**